models/model.js

```diff
@@ -32,4 +30,20 @@ exports.selectArticleById = (id) => {
     .then((results) => {
         return results.rows
     })
-}
\ No newline at end of file
+}
+exports.selectArticles = (topic) => {
+    if (topic!=undefined)
+    {
+    return db.query('SELECT articles.*, COUNT(comment_id)AS comment_count FROM articles LEFT JOIN comments ON comments.article_id =  articles.article_id where articles.topic=$1 GROUP BY articles.article_id ORDER BY articles.created_at desc',[topic])

```

Comment:
This is not very dry. Could we not just amend the query string and add in the where clause if the there is a topic?